src/resources/Question.ts

```diff
@@ -0,0 +1,471 @@
+import { CompoundConditional } from './CompoundConditional'
+
+export type Question0FieldsPreferred = 'none' | 'optional' | 'required'
+
+export interface Question0Fields {
+  /**
+   * Indicates whether the given sub-field should be displayed or not, and if so,
+   * whether a response is optional or required.
+   */
+  preferred: Question0FieldsPreferred
+}
+
+export interface Question0 {
+  type: 'question'
+  question_type: 'name'
+  /**
+   * Unique key of this question within the form. Used in responses and conditional
+   * logic.
+   */
+  key: string
+  /**
+   * Display title of this question in the form, i.e. the question text itself.
+   */
+  title: string
+  /**
+   * Whether a response to this question is required or optional. Questions that are
+   * hidden by conditional rules are never required to have a response.
+   */
+  required: boolean
+  /**
+   * A conditional statement - if it evaluates to true, this question will be
+   * displayed, if false the question will not be displayed and no response will be
+   * recorded for it.
+   */
+  conditional?: CompoundConditional
+  /**
+   * Indicates that the response from this question should be used to populate or
+   * update the member's name.
+   */
+  destination?: 'member.name'
+  /**
+   * Indicates which fields are optional, required, or should not be shown at all.
+   */
+  fields: Question0Fields
+}
+
+export type Question1InputType = 'date_picker' | 'date_entry'
+
+export interface Question1 {
```

Comment:
Comment: we could improve this automated naming by naming all of the form config sub-codecs, and if we give them a namespace they would end up at e.g. `forms/questions/DateQuestion.ts`.